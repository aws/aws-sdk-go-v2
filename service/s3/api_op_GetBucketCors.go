// Code generated by smithy-go-codegen DO NOT EDIT.

package s3

import (
	"context"
	awsmiddleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
	"github.com/aws/aws-sdk-go-v2/aws/retry"
	"github.com/aws/aws-sdk-go-v2/aws/signer/v4"
	"github.com/aws/aws-sdk-go-v2/service/s3/types"
	smithy "github.com/awslabs/smithy-go"
	"github.com/awslabs/smithy-go/middleware"
	smithyhttp "github.com/awslabs/smithy-go/transport/http"
)

// Returns the cors configuration information set for the bucket.  <p> To use this
// operation, you must have permission to perform the s3:GetBucketCORS action. By
// default, the bucket owner has this permission and can grant it to others.</p>
// <p> For more information about cors, see <a
// href="https://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html"> Enabling
// Cross-Origin Resource Sharing</a>.</p> <p>The following operations are related
// to <code>GetBucketCors</code>:</p> <ul> <li> <p> <a>PutBucketCors</a> </p> </li>
// <li> <p> <a>DeleteBucketCors</a> </p> </li> </ul>
func (c *Client) GetBucketCors(ctx context.Context, params *GetBucketCorsInput, optFns ...func(*Options)) (*GetBucketCorsOutput, error) {
	stack := middleware.NewStack("GetBucketCors", smithyhttp.NewStackRequest)
	options := c.options.Copy()
	for _, fn := range optFns {
		fn(&options)
	}
	addawsRestxml_serdeOpGetBucketCorsMiddlewares(stack)
	awsmiddleware.AddRequestInvocationIDMiddleware(stack)
	smithyhttp.AddContentLengthMiddleware(stack)
	AddResolveEndpointMiddleware(stack, options)
	v4.AddComputePayloadSHA256Middleware(stack)
	retry.AddRetryMiddlewares(stack, options)
	addHTTPSignerV4Middleware(stack, options)
	awsmiddleware.AddAttemptClockSkewMiddleware(stack)
	addClientUserAgent(stack)
	smithyhttp.AddErrorCloseResponseBodyMiddleware(stack)
	smithyhttp.AddCloseResponseBodyMiddleware(stack)
	addOpGetBucketCorsValidationMiddleware(stack)
	stack.Initialize.Add(newServiceMetadataMiddleware_opGetBucketCors(options.Region), middleware.Before)
	addUpdateEndpointMiddleware(stack, options)

	for _, fn := range options.APIOptions {
		if err := fn(stack); err != nil {
			return nil, err
		}
	}
	handler := middleware.DecorateHandler(smithyhttp.NewClientHandler(options.HTTPClient), stack)
	result, metadata, err := handler.Handle(ctx, params)
	if err != nil {
		return nil, &smithy.OperationError{
			ServiceID:     ServiceID,
			OperationName: "GetBucketCors",
			Err:           err,
		}
	}
	out := result.(*GetBucketCorsOutput)
	out.ResultMetadata = metadata
	return out, nil
}

type GetBucketCorsInput struct {
	// The bucket name for which to get the cors configuration.
	Bucket *string
}

type GetBucketCorsOutput struct {
	// A set of origins and methods (cross-origin access that you want to allow). You
	// can add up to 100 rules to the configuration.
	CORSRules []*types.CORSRule

	// Metadata pertaining to the operation's result.
	ResultMetadata middleware.Metadata
}

func addawsRestxml_serdeOpGetBucketCorsMiddlewares(stack *middleware.Stack) {
	stack.Serialize.Add(&awsRestxml_serializeOpGetBucketCors{}, middleware.After)
	stack.Deserialize.Add(&awsRestxml_deserializeOpGetBucketCors{}, middleware.After)
}

func newServiceMetadataMiddleware_opGetBucketCors(region string) awsmiddleware.RegisterServiceMetadata {
	return awsmiddleware.RegisterServiceMetadata{
		Region:        region,
		ServiceID:     ServiceID,
		SigningName:   "s3",
		OperationName: "GetBucketCors",
	}
}
