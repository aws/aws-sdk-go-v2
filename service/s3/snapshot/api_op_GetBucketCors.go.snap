GetBucketCors
	Initialize stack step
		spanInitializeStart
		RegisterServiceMetadata
		legacyEndpointContextSetter
		S3Shared:ARNLookup
		SetLogger
		OperationInputValidation
		spanInitializeEnd
	Serialize stack step
		spanBuildRequestStart
		putBucketContext
		setOperationInput
		ResolveEndpoint
		S3:ProcessARNResource
		S3:ObjectLambdaEndpoint
		serializeImmutableHostnameBucket
		OperationSerializer
		S3:UpdateEndpoint
		S3:RemoveBucketFromPathMiddleware
		isExpressUserAgent
	Build stack step
		ClientRequestID
		ComputeContentLength
		UserAgent
		AddTimeOffsetMiddleware
		RecursionDetection
		spanBuildRequestEnd
	Finalize stack step
		DisableAcceptEncodingGzip
		spanRetryLoop
		Retry
		RetryMetricsHeader
		ResolveAuthScheme
		GetIdentity
		ResolveEndpointV2
		disableHTTPS
		ComputePayloadHash
		SigV4ContentSHA256Header
		setLegacyContextSigningOptions
		Signing
	Deserialize stack step
		AddRawResponseToMetadata
		ErrorCloseResponseBody
		CloseResponseBody
		ResponseErrorWrapper
		S3MetadataRetriever
		OperationDeserializer
		AddTimeOffsetMiddleware
		RecordResponseTiming
		RequestResponseLogger
